RecDP/pyrecdp/primitives/llmutils/sentence_split.py

```diff
@@ -0,0 +1,32 @@
+from pyspark.sql import DataFrame
+from pyspark.sql import Row as SparkRow
+from pyrecdp.models.model_utils import get_model
+from pyrecdp.models.helper_func import get_sentences_from_document
+
+
+class SentenceSplit:
+    def __init__(self, lang: str = 'en'):
+        nltk_model = get_model(lang, model_type="nltk")
+        self.tokenizer = nltk_model.tokenize if nltk_model else None
+
+    def process(self, sample) -> str:
+        """
+        Get sentences from a document.
+
+        :return: document with the sentences separated by '\\\\n'
+
+        Args:
+            sample: document that need to split sentences
+        """
+        return get_sentences_from_document(sample, model_func=self.tokenizer)
+
+
+def sentence_split(dataset: DataFrame, text_column='text') -> DataFrame:
+    def do_split(batch):
+        sentence_plit = SentenceSplit()
+        for row in batch:
+            row_dict = dict(**row.asDict())
```

Comment:
I still a little bit concern about this way of converting row to dict, it will include more data to be pickled from java to spark I think.